source/common/tls/default_tls_certificate_selector.cc

```diff
@@ -206,14 +202,14 @@ DefaultTlsCertificateSelector::findTlsContext(
       }
     }
 
-    // if the client is not ECDSA-capable and the `ctx` is non-ECDSA, then select this `ctx`
-    if (!client_ecdsa_capabilities.has_value() && !ctx.is_ecdsa_) {
+    // If the client is not ECDSA-capable and the `ctx` is non-ECDSA, then select this `ctx`.
+    if (cec_count == 0 && ctx.ec_group_curve_name_ == 0) {
```

Comment:
Please define a constant for `EC_CURVE_INVALID_NID` or similar (value can be zero, or maybe negative one is better). Use that both in initializing `ec_group_curve_name_` in the header, and in the comparison here. It makes the code clearer and more self-documenting.